x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_index/connector/connector_configuration_logic.ts

```diff
@@ -119,13 +127,66 @@ function sortAndFilterConnectorConfiguration(config: ConnectorConfiguration): Co
   );
 }
 
+function validateConnectorConfiguration(config: ConfigEntry[]): ConfigEntry[] {
+  return config.map((configEntry) => {
+    const label = configEntry.label;
+
+    configEntry.validation_errors = [];
+
+    switch (configEntry.type) {
+      case FieldType.INTEGER:
+        if (!validIntInput(configEntry.value)) {
+          configEntry.validation_errors.push(
+            i18n.translate(
+              'xpack.enterpriseSearch.content.indices.configurationConnector.config.invalidInteger',
+              {
+                defaultMessage: '{label} must be an integer.',
+                values: { label },
+              }
+            )
+          );
+        }
+    }
+
+    configEntry.is_valid = configEntry.validation_errors.length > 0 ? false : true;
+
+    return configEntry;
+  });
+}
+
+function validIntInput(value: string | number | boolean | null): boolean {
+  // reject non integers (including floats), but don't validate if empty
+  return (value !== null || value !== '') &&
```

Comment:
This will return true for Infinity, If you type `3/0` as an input it will return true and value would be Infinity. 

Maybe we can use `Number.isSafeInteger()` to make sure it is smaller than BigInt boundries and Integer.

Reply:
Thanks, I wouldn't have thought of this case.